scripts/etc/init.d/rcS

```diff
@@ -533,66 +533,66 @@ fi
+            echo "OK"
+        fi
+        echo -n "  Setting root password... "
+        echo -n "root:$rootpw" | chroot /rootfs /usr/sbin/chpasswd || fail
         echo "OK"
-    else
-        echo "FAILED !"
     fi
 fi
 
 # add user to system
 if [ ! -z "$username" ]; then
-    echo -n "  Creating user... "
-    chroot /rootfs /usr/sbin/adduser "$username" --gecos "" --disabled-password
-    if [ $? -eq 0 ]; then
+    echo -n "  Creating $username user... "
+    chroot /rootfs /usr/sbin/adduser "$username" --gecos "" --disabled-password || fail
+    echo "OK"
+    # add SSH key for user (if provided)
+    if [ ! -z "$user_ssh_pubkey" ]; then
+        echo -n "  Setting $username SSH key... "
+        ssh_dir="/rootfs/home/$username/.ssh"
+        mkdir -p "$ssh_dir" && \
+            echo "$user_ssh_pubkey" > "$ssh_dir/authorized_keys" || fail
         echo "OK"
-        echo -n "  Setting user password... "
-        echo -n $username:$userpw | chroot /rootfs /usr/sbin/chpasswd
-        if [ $? -eq 0 ]; then
-            echo "OK"
-        else
-            echo "FAILED !"
-        fi
-        if [ ! -z "$user_ssh_pubkey" ]; then
-            echo -n "  Setting user SSH key... "
-            ssh_dir="/home/$username/.ssh"
-            mkdir -p "/rootfs$ssh_dir" && \
-                chmod 700 "/rootfs$ssh_dir" && \
-                echo "$user_ssh_pubkey" > "/rootfs$ssh_dir/authorized_keys"
-            if [ $? -eq 0 ]; then
-                echo "OK"
-                echo -n "  Setting owner on SSH key... "
-                chroot /rootfs /bin/chown -R "$username:$username" "$ssh_dir"
-                if [ $? -eq 0 ]; then
-                    echo "OK"
-                else
-                    echo "FAILED !"
-                fi
-            else
-                echo "FAILED !"
-            fi
-        fi
-        if [ "$user_is_admin" = "1" ]; then
-            echo -n "  Adding $username to sudo group... "
-            chroot /rootfs /usr/sbin/usermod -aG sudo "$username"
+        echo -n "  Setting owner as $username on SSH directory... "
+        chroot /rootfs /bin/chown -R "$username:$username" "/home/$username/.ssh" || fail
+        echo "OK"
+        echo -n "  Setting permissions on $username SSH directory... "
+        chmod -R go-rwx "$ssh_dir" || fail
+        echo "OK"
+    fi
+    if [ ! -z "$user_passwd" ]; then
+        echo -n "  Setting $username password... "
+        echo -n "$username:$userpw" | chroot /rootfs /usr/sbin/chpasswd || fail
+        echo "OK"
+    fi
+    if [ "$user_is_admin" = "1" ]; then
+        echo -n "  Adding $username to sudo group... "
+        chroot /rootfs /usr/sbin/usermod -aG sudo "$username" || fail
+        echo "OK"
+        if [ -z "$user_passwd" ]; then
+            echo -n "  Setting $username to sudo without a password... "
+            echo -n "%sudo         ALL = (ALL) NOPASSWD: ALL" > /rootfs/etc/sudoers.d/01_sudo || fail
```

Comment:
I've just created a fresh VM with jessie and during installation it asks (and suggest you do) for a root password. 
If you do leave it blank, sudo is indeed setup, but **NOT** with the `NOPASSWD` switch.
And not using `root` and setting up the user without a password is NOT allowed.
```
diederik@deb-jessie-vm:~$ sudo cat etc/sudoers
#
# This file MUST be edited with the 'visudo' command as root.
#                         
# Please consider adding local content in /etc/sudoers.d/ instead of
# directly modifying this file.
#                         
# See the man page for details on how to write a sudoers file.
#                         
Defaults        env_reset
Defaults        mail_badpass
Defaults        secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

# Host alias specification

# User alias specification

# Cmnd alias specification

# User privilege specification
root    ALL=(ALL:ALL) ALL

# Allow members of group sudo to execute any command
%sudo   ALL=(ALL:ALL) ALL

# See sudoers(5) for more information on "#include" directives:

#includedir /etc/sudoers.d
```